RealmSwift/Tests/MapTests.swift

```diff
@@ -547,6 +547,48 @@ class MapTests: TestCase {
         }
     }
 
+    func testKeyedSortable() {
+        let map = createMap()
+
+        map["key"] = SwiftStringObject(value: ["apples"])
+        map["key2"] = SwiftStringObject(value: ["bananas"])
+        map["key3"] = SwiftStringObject(value: ["cockroach"])
+
+        if map.realm != nil {
+            let results2: Results<SwiftStringObject?> = map.sorted(by: \.stringCol,
+                                                                   ascending: true)
+            XCTAssertEqual(results2.count, 3)
+            XCTAssertEqual(results2[0]?.stringCol, "apples")
+            XCTAssertEqual(results2[1]?.stringCol, "bananas")
+            XCTAssertEqual(results2[2]?.stringCol, "cockroach")
+        } else {
+            assertThrows(map.sorted(by: \.stringCol, ascending: true))
+        }
+    }
+
+    func testKeyedAggregatable() {
+        let map = realm.create(SwiftMapPropertyObject.self, value: [])
```

Comment:
won't this always be managed?

Reply:
Ah right, should add in an add and not create it through the Realm. Though I suppose it's technically not managed until the commit.